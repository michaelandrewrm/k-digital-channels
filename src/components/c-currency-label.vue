<template>
	<div class="c-currency-label">
		<span class="c-currency-label__label text-l-book" v-if="startsWithCurrency">
			{{ currency }}
		</span>
		<span class="c-currency-label__label text-xxl-book">{{ number }}</span>
		<span class="c-currency-label__label text-l-book" v-if="!startsWithCurrency">
			{{ currency }}
		</span>
	</div>
</template>

<script>
export default {
	name: 'c-currency-label',

	props: {
		balance: { type: Object },
	},

	computed: {
		formatted({ balance: { amount, currency } }) {
			return this.$n(amount, {
				minimumFractionDigits: 2,
				style: 'currency',
				currency: currency.id,
				currencyDisplay: 'symbol',
				useGrouping: true,
			});
		},

		number({ balance: { amount } }) {
			return this.$n(amount, { minimumFractionDigits: 2 });
		},

		currency({ formatted, number }) {
			return formatted.replace(number, '').trim();
		},

		startsWithCurrency({ formatted, currency }) {
			return formatted.startsWith(currency);
		},
	},
};
</script>

<style lang="scss" scoped>
.c-currency-label {
	display: flex;
	align-items: center;
	justify-content: center;
}

.c-currency-label__label {
	margin: 0 3px;
}
</style>
