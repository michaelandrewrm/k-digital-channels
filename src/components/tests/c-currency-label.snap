// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-currency-label.vue renders corrently 1`] = `
<div class="c-currency-label">
  <!---->
  <span class="c-currency-label__label text-xxl-book">142,42</span>
  <span class="c-currency-label__label text-l-book">
    â‚¬
  </span>
</div>
`;
